cpp/include/Ice/StreamableTraits.h

```diff
@@ -13,373 +13,277 @@ namespace Ice
     /// typedef using SFINAE.
     /// @tparam T The type to check
-    /// @headerfile Ice/Ice.h
     template<typename T> struct IsContainer
     {
-        /// @private
         template<typename C> static char test(typename C::iterator*) noexcept;
 
-        /// @private
         template<typename C> static long test(...) noexcept;
 
-        /// `true` when @p T is a container, otherwise `false`.
-        static const bool value = sizeof(test<T>(nullptr)) == sizeof(char);
+        static constexpr bool value = sizeof(test<T>(nullptr)) == sizeof(char);
     };
 
+    /// @private
     /// Determines whether the provided type is a map. The implementation checks if there is a @p T::mapped_type typedef
     /// using SFINAE.
     /// @tparam T The type to check
-    /// @headerfile Ice/Ice.h
     template<typename T> struct IsMap
     {
-        /// @private
         template<typename C> static char test(typename C::mapped_type*) noexcept;
 
-        /// @private
         template<typename C> static long test(...) noexcept;
 
-        /// `true` when @p T is a map, otherwise `false`.
-        static const bool value = IsContainer<T>::value && sizeof(test<T>(nullptr)) == sizeof(char);
+        static constexpr bool value = IsContainer<T>::value && sizeof(test<T>(nullptr)) == sizeof(char);
     };
 
-    /// Base traits template. Types with no specialized trait use this trait.
-    /// @tparam T The type.
-    /// @tparam Enabler The specialization enabler.
+#ifdef ICE_DOXYGEN
+    /// Provides traits for a type that can be marshaled or unmarshaled to/from a stream of bytes using the Slice
+    /// encoding. This template is specialized for each type that can be marshaled/unmarshaled.
+    /// @tparam T The type to provide the traits for.
+    /// @tparam Enabler A type used to enable a partial specialization for several types. It should not be used in the
+    /// partial specialization itself.
+    /// @remark Streamable traits for enumeration types provide two additional traits: `minValue` and `maxValue`.
     /// @headerfile Ice/Ice.h
     template<typename T, typename Enabler = void> struct StreamableTraits
     {
         /// The category trait, used for selecting the appropriate StreamHelper.
-        static const StreamHelperCategory helper = StreamHelperCategoryUnknown;
-
-        //
-        // When extracting a sequence<T> from a stream, we can ensure the
-        // stream has at least StreamableTraits<T>::minWireSize * size bytes
-        // For containers, the minWireSize is 1 (just 1 byte for an empty container).
-        //
-        // static const int minWireSize = 1;
-
-        //
-        // Is this type encoded on a fixed number of bytes?
-        // Used only for marshaling/unmarshaling optional data members and parameters.
-        //
-        // static const bool fixedLength = false;
+        static constexpr StreamHelperCategory helper;
```

Comment:
While this is not really valid C++ (no value is assigned to these constexpr), doxygen is happy with it.